pyannotate_tools/fixes/fix_annotate_json.py

```diff
@@ -82,22 +82,22 @@ def get_init_file(dir):
             return f
     return None
 
-def get_funcname(name, node):
-    # type: (Leaf, Node) -> Text
+def get_funcname(node):
+    # type: (Optional[Node]) -> Text
     """Get function name by the following rules:
 
     - function -> function_name
     - instance method -> ClassName.function_name
     """
-    funcname = name.value
-    if node.parent and node.parent.parent:
-        grand = node.parent.parent
-        if grand.type == syms.classdef:
-            grandname = grand.children[1]
-            assert grandname.type == token.NAME, repr(name)
-            assert isinstance(grandname, Leaf)  # Same as previous, for mypy
-            funcname = grandname.value + '.' + funcname
-    return funcname
+    components = []  # type: List[str]
+    while node:
+        if node.type in (syms.classdef, syms.funcdef):
```

Comment:
Should this do something about async defs?

Reply:
It doesn't need to, since `async_funcddef` contains a `funcdef`, and that's what gets passed in. I'll add a test, since this means that "async methods" are one of the other things fixed by this.